controllers/controllers.js

```diff
@@ -7,12 +7,30 @@ exports.getAllEndpoints = (req, res, next) => {
 
 exports.getAllTopics = (req, res, next) => {
     selectAllTopics()
-    .then(({rows}) => {
+    .then(({ rows }) => {
         const topics = rows
 
         res.status(200).send({topics})
     })
     .catch((err) => {
         next(err)
     })
+}
+
+exports.getArticleById = (req, res, next) => {
+    const { article_id } = req.params
+    selectArticleById(article_id)
+    .then(({ rows }) => {
+        if (rows.length === 0) {
```

Comment:
Another way to do this is to check this in the model and reject the promise there with this error object so that the controller is only concerned with communicating directly with the client (In this case we'd invoke next within the catch block here). Database related checks/processing should happen within the model